temporalcli/commandsgen/parse.go

```diff
@@ -45,6 +47,16 @@ type (
 		Options                []Option `yaml:"options"`
 		OptionSets             []string `yaml:"option-sets"`
 		Docs                   Docs     `yaml:"docs"`
+		Index                  int
```

Comment:
How many of these fields are required for docs gen? I don't see `Index` or `SubCommandName` or any of that used there. Are all of these fields needed to generate docs from the existing structure?

Reply:
`Index` was needed to [set other derivative fields](https://github.com/temporalio/cli/pull/691/files/68f575f383e943323d11c0c71f6762eec38aebea#diff-cc2ae8272d97aad62d713b302b77891db4f3e1c9cbf504e4aabcc2e49c46badaR247) in `parse.go`.

`SubCommandName` was used for auto-generated keywords to match the existing docs, but if we're not going to auto-generate keywords we can probably remove this one.

Reply:
> across multiple passes so passing it around would be cumbersome.

I think it is best to use contextual state when recursing over updating the external state of the object as temporary storage. Or maybe even better, derive it live as needed during docs gen.

Reply:
> How many of these fields are required for docs gen?

Unused fields have been removed.

> Or maybe even better, derive it live as needed during docs gen.

What would you propose for [deriving `isLeafCommand()`](https://github.com/temporalio/cli/pull/691/files#diff-204e1216d4ad67aaaa69283810195e4b426bb9ba834c3ec336a8c8f2fdb1553dR84) during the single pass `docs.go` gen? Scanning the full list of commands for children for each subcommand with `slices.IndexFunc` or similar, since performance isn't a concern?

> I think it is best to use contextual state when recursing over updating the external state of the object as temporary storage.

So use a lookup map for the index or scan with `slices.IndexFunc`?

Reply:
I see, the `code.go` is just [scanning for subcommands](https://github.com/temporalio/cli/blob/main/temporalcli/commandsgen/code.go#L182) for each command.

If that's the approach we're using, happy to make it consistent.

Reply:
I am a fan of it, but I won't push too hard on this code since it's build-time tooling.

> Scanning the full list of commands for children for each subcommand with slices.IndexFunc or similar, since performance isn't a concern?

Definitely not a concern for this use case

> So use a lookup map for the index or scan with slices.IndexFunc?

Probably only need `slices.Index` if you have a pointer, but sure.

Reply:
> I am a fan of it, but I won't push too hard on this code since it's build-time tooling.

No worries, updated to use derived state to keep the model small.